src/cargo/core/compiler/mod.rs

```diff
@@ -139,8 +141,21 @@ fn compile<'a, 'cfg: 'a>(
             };
             work.then(link_targets(cx, unit, false)?)
         } else {
+            let work = if cx.bcx.build_config.cache_messages()
+                && cx.bcx.show_warnings(unit.pkg.package_id())
```

Comment:
How come `show_warnings` is included here? I would imagine that if `show_warnings` were false we'd already in general have an empty cache, but there's been occasional warnings over time which bypass lint settings and warn even for transitive dependencies, and if we want to replay everything we may not want to suppress them.

Reply:
Hm this is a good point, and it sounds pretty unfortunate. One thing I'm worried about is that rustc may decide that not all warnings fall under `--cap-lints`. For example "please urgently fix this dependency since it will break soon" may be issued for upstream dependencies anyway. In that sense I don't think we can move `--cap-lints` into Cargo (not without more information in the json blob at least) easily.

That being said not spewing warnings on `-vv` followed by `cargo build` sounds good. The "failure case" of "less warnings on `cargo build` followed by -`vv`" seems no worse than today.

This all in all sounds like a good thing to note on a tracking issue for this :)